docker/src/main/java/brooklyn/location/docker/DockerContainerLocation.java

```diff
@@ -168,20 +171,57 @@ public HostAndPort getSocketEndpointFor(Cidr accessor, int privatePort) {
 
     @Override
     public int execScript(Map<String,?> props, String summaryForLogging, List<String> commands, Map<String,?> env) {
-        Iterable<String> filtered = Iterables.filter(commands, DockerCallbacks.FILTER);
-        for (String commandString : filtered) {
-            parseDockerCallback(commandString);
+        if(getOwner().config().get(DockerContainer.DOCKER_USE_EXEC)) {
+            Map<String,?> nonPortProps = Maps.filterKeys(props, Predicates.not(Predicates.containsPattern("port")));
+            SshMachineLocation host = getOwner().getDockerHost().getDynamicLocation().getMachine();
+            return host.execCommands(nonPortProps, summaryForLogging, getExecScript(commands, env));
+        } else {
+            Iterable<String> filtered = Iterables.filter(commands, DockerCallbacks.FILTER);
+            for (String commandString : filtered) {
+                parseDockerCallback(commandString);
```

Comment:
needs to happen for both sides of the conditional, this is where `docker commit` or `docker push` happens